consensus/parlia/parlia.go

```diff
@@ -327,9 +327,26 @@ func (p *Parlia) verifyHeader(chain consensus.ChainHeaderReader, header *types.H
 	}
 	number := header.Number.Uint64()
 
-	// Don't waste time checking blocks from the future
+	// According to BEP-?, after Bohr fork, an in-turn validator is allowed to broadcast
+	// a mined block earlier but not earlier than its parent's timestamp when the block is ready .
 	if header.Time > uint64(time.Now().Unix()) {
-		return consensus.ErrFutureBlock
+		if !chain.Config().IsBohr(header.Number) || header.Difficulty.Cmp(diffInTurn) != 0 {
+			return consensus.ErrFutureBlock
+		}
+		var parent *types.Header
+		if len(parents) > 0 {
+			parent = parents[len(parents)-1]
+		} else {
+			parent = chain.GetHeader(header.ParentHash, number-1)
+		}
+
+		if parent == nil || parent.Number.Uint64() != number-1 || parent.Hash() != header.ParentHash {
+			return consensus.ErrUnknownAncestor
+		}
+
+		if parent.Time > uint64(time.Now().Unix()) {
+			return consensus.ErrBroadcastTooEarly
+		}
```

Comment:
If we have a hard fork here, why don't we implement the early broadcast logic in this PR?